lib/collect.js

```diff
@@ -71,11 +73,39 @@ module.exports.chargeToken = function (options, next) {
       businessId: options.businessId,
     },
     amounts: amounts,
-    fundingSource: {
-      cardToken: options.token,
-    },
   };
 
+  // Choosing funding source
+  if (options.nonce && options.token) {
+    var hasErr = helpers.error({
+      name: 'InvalidArgumentError',
+      message: 'Invalid params: Request should contain either nonce or token',
+      statusCode: 400
+    });
+    return next(hasErr);
+  }
+
+  // AX-3758: charge using nonce as funding source
+  if (options.nonce) {
+    var hasErr = helpers.hasKeys(options, ["tokenize"]);
+    if (hasErr) {
+      return next(hasErr);
+    }
+    body.fundingSource = {
+      nonce: options.nonce,
```

Comment:
it should be either of them, does this one mean both ?

Reply:
we cant have both, it will be confusing.

Reply:
either

Reply:
line 79 does the checking if there exist both